### ISSUE TITLE
Output exa results differently in scripting environment with grid options

### ISSUE THREAD

**Issue by Zhengqbbb** [ogham/exa#1083]
If exa does something unexpected, or its output looks wrong, or it displays an error on the screen, or if it outright crashes, then please include the following information in your report:

- The version of exa being used (`exa --version`)
- The command-line arguments you are using
- Your operating system and hardware platform

If it’s a crash, please include the full text of the crash that gets printed to the screen. If you’re seeing unexpected behaviour, a screenshot of the issue will help a lot.

---
- `exa --version`: v0.10.1

In scripting environment. I hope to be able to record the output of exa completely, but as you can see, it is output in `oneline` form, but I have specified my options `--grid` but it doesn't seem to take effect.

<img width="637" alt="image" src="https://user-images.githubusercontent.com/40693636/177933253-67502ebe-c60a-47e5-b277-627d2b61b147.png">

```sh
```

**Comment by dylan-tock** [comment]
Experiencing the same issue.

![image](https://user-images.githubusercontent.com/65966559/235365416-56e0bd90-a5a6-4976-9cd9-f8fb2f4feb6d.png)

And here's the script, output, and system/software details:

```
workstation> cat a.sh
#!/usr/bin/zsh

printf "== Inside subshell\n"
a="$(exa --color=always --grid)"
printf "%s\n" "$a"

printf "%s\n" ""

printf "== Plain command\n"
exa --color=always --grid

printf "%s\n" ""

printf "== Inside subshell with COLUMNS variable explicitly set\n"
a="$(COLUMNS="${COLUMNS}" exa --color=always --grid)"
printf "%s\n" "$a"

printf "%s\n" ""

workstation> ./a.sh
== Inside subshell
a
a.sh
b

== Plain command
a  a.sh  b

== Inside subshell with COLUMNS variable explicitly set
a  a.sh  b

workstation> exa --version
exa - list files on the command-line
v0.10.1 [+git]
https://the.exa.website/
workstation> uname -a
Linux workstation 5.15.0-1032-gcp #40~20.04.1-Ubuntu SMP Tue Apr 11 02:49:52 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
workstation> zsh --version
zsh 5.8 (x86_64-ubuntu-linux-gnu)
workstation>
```

This root cause might be the same as what's reported in [issue 990](https://github.com/ogham/exa/issues/990).

**Comment by ariasuni** [comment]
Does it still happen with latest master, or with [the active fork eza)[https://github.com/eza-community/eza]?

**Comment by ariasuni** [comment]
Hum thinking about it, I don’t think it’s wrong. I believe In a script, exa doesn’t know about the number of columns of the terminal, or because it’s easier to treat output line by line in scripts. Besides, it’s also the behavior of `ls`, so in that regards it can be considered correct, at least by default.

**Comment by Zhengqbbb** [comment]
> @ariasuni Hum thinking about it, I don’t think it’s wrong. I believe In a script, exa doesn’t know about the number of columns of the terminal, or because it’s easier to treat output line by line in scripts. Besides, it’s also the behavior of `ls`, so in that regards it can be considered correct, at least by default.

I agree with your point. In fact, it's normal to handle "array" behavior line by line in a script environment. 

However, In my opinion **the CLI provides this option** and a user uses it, but the resulting behavior is not as expected, it could lead to the script becoming ineffective and unreasonable.

**Comment by ariasuni** [comment]
Ah I see, so when passing `--grid` explicitly, you’d like exa to display as a grid even in subshells. It seems to be possible with [any_terminal_size](https://crates.io/crates/any_terminal_size), but it’s not trivial in the code to determine if the flag has been passed explicitly or not. Maybe I could implement this in [eza](https://github.com/eza-community/eza) when [this PR](https://github.com/eza-community/eza/pull/197) is merged.
